lib/pbench/agent/tool_data_sink.py

```diff
@@ -186,6 +190,408 @@ def terminate(self):
+            try:
+                os.mkdir(".temp")
+                self.logger.debug(
+                    "Created folder :{}".format(
+                        os.path.join(self.benchmark_run_dir, ".temp")
+                    )
+                )
+            except Exception:
+                self.logger.debug(
+                    "Folder :{} already exists".format(os.path.join(".temp"))
+                )
+
+            try:
+                os.mkdir(os.path.join(".temp", host))
+                self.logger.debug(
+                    "Created folder :{}".format(os.path.join(".temp", host))
+                )
+            except Exception:
+                self.logger.debug(
+                    "Folder :{} already exists".format(os.path.join(".temp", host))
+                )
+
+            try:
+                for tool in self.host_tools_dict[host]:
+
+                    try:
+                        string = ""
+                        string += "#pmlogconf-setup 2.0\n"
+                        string += "ident   " + self.mapping[tool]["ident"] + "\n"
+                        string += (
+                            "probe   "
+                            + self.mapping[tool]["probe"]
+                            + " ? include : exclude\n\n"
+                        )
+                        for _, metric in enumerate(self.mapping[tool]["metrics"]):
+                            string += "   " + metric + "\n"
+
+                        self.logger.debug(
+                            "%s.summary output for host %s :\n%s", host, host, string
+                        )
+                        with open(
+                            os.path.join(".temp", host, tool + ".summary"), "w"
+                        ) as file:
+                            file.write(string)
+                    except KeyError:
+                        self.logger.error("tool:%s does not exist in mapping", tool)
+
+                # code to build pmlogger configuration file
+                self.logger.info("building pmlogger configuration file")
+                command = "pmlogconf -c -h {} -d {} {}.config ".format(
+                    host, os.path.join(".temp", host), host
+                )
+                self.logger.debug("pmlogconf command:%s", command)
+                out = subprocess.Popen(
+                    [command],
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.STDOUT,
+                    shell=True,
+                )
+                stdout, stderr = out.communicate()
+                self.logger.debug("stdout output:%s", stdout.decode())
+
+                if (
+                    stdout.decode().strip()
+                    != f'Creating config file "{host}.config" using default settings ...\n.'
+                ):
+                    # pmlogger failed to stop. Log it
+                    self.logger.error("not able to build pmlogconf file")
```

Comment:
We can't check the exit status? Judging success by matching output strings is always risky.

(Not to mention that this module has now adding template string formatting to %s formatting and `.format` formatting, making the full sweep of Python string formatting styles and making the whole thing look just a bit more messy. I'm assuming this is an inherited issue, but it'd be nice to clean it up.)